services/management/ia/rules_service.go

```diff
@@ -255,11 +256,100 @@ func (s *RulesService) ListAlertRules(ctx context.Context, req *iav1beta1.ListAl
 		return nil, status.Errorf(codes.FailedPrecondition, "%v.", services.ErrAlertingDisabled)
 	}
 
-	res, err := s.getAlertRules()
+	const defaultPageSize = 20
+
+	var pageIndex, pageSize int
+	if req.Page == nil {
+		pageSize = defaultPageSize
+	} else {
+		pageIndex = int(req.Page.Index)
+		pageSize = int(req.Page.Size)
+	}
```

Comment:
IMHO without `else` it's easier to read.

```suggestion
	pageIndex = 0 // first page
	pageSize =  defaultPageSize
	if req.Page != nil {
		pageIndex = int(req.Page.Index)
		pageSize = int(req.Page.Size)
	}
```

Reply:
for sure